.github/actions/download-engine/action.yml

```
@@ -0,0 +1,91 @@
+name: "Download VOICEVOX ENGINE"
+description: |
+  VOICEVOX ENGINEをダウンロードし、指定したディレクトリに展開する。
+inputs:
+  version:
+    description: "VOICEVOX ENGINEのバージョン。latest（デフォルト）、preview、バージョン番号（例：0.14.4）で指定できる。"
+    required: false
+    default: "latest"
+  dest:
+    description: "VOICEVOX ENGINEを展開するディレクトリ。"
+    required: true
+  target:
+    description: "ダウンロードする対象。デフォルトは各OSのCPU版。"
+    required: false
+    default: ""
+
+outputs:
+  run_path:
+    description: "run.exe、またはrunのパス。"
+    value: ${{ steps.result.outputs.run_path }}
+  version:
+    description: "VOICEVOX ENGINEのバージョン。"
+    value: ${{ steps.result.outputs.version }}
+runs:
+  using: "composite"
+  steps:
+    - name: Setup tempdir
+      shell: bash
+      run: |
+        TEMPDIR=$(echo '${{ runner.temp }}' | sed -e 's_\\_/_g')
+        echo TEMPDIR=$TEMPDIR >> $GITHUB_ENV
+        mkdir -p $TEMPDIR
+
+        DEST=$(echo '${{ inputs.dest }}' | sed -e 's_\\_/_g')
+        echo DEST=$DEST >> $GITHUB_ENV
+
+    - name: Get version
+      shell: bash
+      run: |
+        curl -s https://api.github.com/repos/voicevox/voicevox_engine/releases \
+            -H 'authorization: Bearer ${{ github.token }}' \
+            -H 'content-type: application/json' > $TEMPDIR/releases.json
+
+        if [ "${{ inputs.version }}" = "latest" ]; then
+          cat $TEMPDIR/releases.json | jq -er '[.[] | select(.prerelease == false)][0]' > $TEMPDIR/target.json
+        elif [ "${{ inputs.version }}" = "preview" ]; then
+          cat $TEMPDIR/releases.json | jq -er '[.[] | select(.prerelease == true)][0]' > $TEMPDIR/target.json
+        else
+          cat $TEMPDIR/releases.json | jq -er '[.[] | select(.tag_name == "${{ inputs.version }}")][0]' > $TEMPDIR/target.json
+        fi
+    - name: Download and Extract
+      shell: bash
+      run: |
+        if [ "${{ inputs.target }}" = "" ]; then
+          OS="${{ runner.os }}"
+          TARGET="${OS,,}-cpu"
```

ちょっと難解なのでコメントでメモしておくのはどうでしょう。
```suggestion
          TARGET="${OS,,}-cpu" # 小文字にする
```